src/core/state.go

```diff
@@ -749,6 +749,10 @@ func (state *BuildState) ensureDownloaded(target *BuildTarget) {
 
 // QueueTarget adds a single target to the build queue.
 func (state *BuildState) QueueTarget(label, dependent BuildLabel, rescan, forceBuild bool) error {
+	return state.queueTarget(label, dependent, rescan, forceBuild, forceBuild)
```

Comment:
Why are we passing forceBuild for both here? Shouldn't this be tartget.NeededForSubinclude?

Reply:
No, because this is the thing that sets NeededForSubinclude. Breaking it apart this way means only the "outside" call gets this set, not any of its dependencies.

Reply:
Well it can also be set to true based on `dependent.IsAllTargets()` which seems odd but i guess that's a way of detecting if it's a subinclude?